src/common/libidset/idset.h

```diff
@@ -84,21 +91,31 @@ bool idset_test (const struct idset *idset, unsigned int id);
  */
 unsigned int idset_first (const struct idset *idset);
 
-/* Return the next id after 'prev' in the idset.
- * Returns IDSET_INVALID_ID if prev is the last id.
+/* Return the next id after 'id' in the idset.
+ * Returns IDSET_INVALID_ID if 'id' is the last id.
  */
-unsigned int idset_next (const struct idset *idset, unsigned int prev);
+unsigned int idset_next (const struct idset *idset, unsigned int id);
 
 /* Returns the last id in the idset.
  * Returns IDSET_INVALID_ID if the idset is empty.
  */
 unsigned int idset_last (const struct idset *idset);
 
+/* Return the previous id before 'id' in the idset.
+ * Returns IDSET_INVALID_ID if 'id' is the first id.
+ */
+unsigned int idset_prev (const struct idset *idset, unsigned int id);
```

Comment:
Just a thought: If there is no current use case for idset_prev() it might make sense to leave it out for now in case the veb implementation is replaced in the future due to the poor space efficiency of vEB trees (discussed in #5287) and the replacement has trouble with a find previous operation?

Reply:
I think that was the justification for leaving it out originally.  However, all of these possible replacements have both predecessor and successor operations (well croaring has an iterator that works forward and reverse - not sure how those would map to idsets simpler interfaces)

https://en.wikipedia.org/wiki/X-fast_trie

https://en.wikipedia.org/wiki/Y-fast_trie

https://github.com/RoaringBitmap/CRoaring/blob/59b740fa5c6dafb47bfc0fbc326b0080061a2441/include/roaring/roaring.h#L925C1-L930


and it seemed like a hole in the API to have first, next, and last but no prev.

However, it's not a big deal - we could leave it out until we need it if you like.

Reply:
No, it is fine, it was just a thought since I didn't see any users.

Reply:
BTW, the Wikipedia article for vEB trees now also mentions [Fusion tree](https://en.wikipedia.org/wiki/Fusion_tree) as another alternative with better space efficiency and better search than vEB (for large _w_). However, we're not dealing with large _w_ (and likely won't?) here so this discussion is largely academic (and sorry for the tangent!)

Reply:
That's OK, good discussion!

Reply:
Fusion tree also implements both predecessor and successor, so that's good.